flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/plan/util/FlinkRelOptUtil.scala

```diff
@@ -211,4 +247,62 @@ object FlinkRelOptUtil {
     }
   }
 
+  /**
+    * Merge two MiniBatchInterval as a new one.
+    *
+    * The Merge Logic:  MiniBatchMode: (R: rowtime, P: proctime, N: None), I: Interval
+    * Possible values:
+    * - (R, I = 0): operators that require watermark (window excluded).
+    * - (R, I > 0): window / operators that require watermark with minibatch enabled.
+    * - (P, I > 0): unbounded agg with minibatch enabled.
+    * - (N, I = 0): no operator requires watermark, minibatch disabled
+    * ------------------------------------------------
+    * |    A        |    B        |   merged result
+    * ------------------------------------------------
+    * | R, I_1 == 0 | R, I_2      |  R, gcd(I_1, I_2)
+    * ------------------------------------------------
+    * | R, I_1 == 0 | P, I_2      |  R, I_2
+    * ------------------------------------------------
+    * | R, I_1 > 0  | R, I_2      |  R, gcd(I_1, I_2)
+    * ------------------------------------------------
+    * | R, I_1 > 0  | P, I_2      |  R, I_1
+    * ------------------------------------------------
+    * | P, I_1      | R, I_2 == 0 |  R, I_1
+    * ------------------------------------------------
+    * | P, I_1      | R, I_2 > 0  |  R, I_2
+    * ------------------------------------------------
+    * | P, I_1      | P, I_2 > 0  |  P, I_1
+    * ------------------------------------------------
+    */
+  def mergeMiniBatchInterval(
```

Comment:
Could you defer this method to when we introduce mini batch for streaming? I think this is not a blocker for other features you want introduce for this pr.

Reply:
ok